src/Network/Wai/Handler/Hal.hs

```diff
@@ -152,6 +147,27 @@ defaultOptions =
         _ -> True
     }
 
+-- | A variant of 'run' with configurable 'Options'. Useful if you
+-- just want to override the 'binaryMediaTypes' setting but don't need
+-- the rest of 'runWithContext''s features.
+--
+-- @since 0.4.0.0
+runWithOptions ::
+  (MonadIO m) =>
+  -- | Configuration options. 'defaultOptions' provides sensible defaults.
+  Options ->
+  Wai.Application ->
+  HalRequest.ProxyRequest HalRequest.NoAuthorizer ->
+  m HalResponse.ProxyResponse
+runWithOptions opts app req =
+  liftIO $ do
+    waiReq <- toWaiRequest opts req
+    responseRef <- IORef.newIORef Nothing
+    Wai.ResponseReceived <- app waiReq $ \waiResp ->
+      Wai.ResponseReceived <$ IORef.writeIORef responseRef (Just waiResp)
+    Just waiResp <- IORef.readIORef responseRef
+    fromWaiResponse defaultOptions waiResp
```

Comment:
Should this pass `opts` rather than `defaultOptions`?

Reply:
Yes, fixed.